ci/infra/testrunner/tests/conftest.py

```diff
@@ -94,9 +94,13 @@ def platform(conf, target):
 @pytest.fixture
 def setup(request, platform, skuba):
     def cleanup():
-        platform.gather_logs()
+        # if platform was not allocated, gather_logs may fail. Ignore.
+        try:
+            platform.gather_logs()
+        except:
+            pass
```

Comment:
Nit:

In general I'd be more in favor of structures like
```
try:
  platform.gather_logs()
finally:
  platform.cleanup()
```

This way we still have the chance of adding some exception handling being certain that the  `platform.cleanup` will be called. 

There is not much difference on this specific case, thus I am also fine with the current code if you prefer it.